### ISSUE TITLE
shoule Myocardium Motion based on short aixs of CTA?

### ISSUE THREAD

**Issue by mayayayahong** [shaharzuler/CardioSpectrum#1]
hello shaharzuler, thank you for your release, I am very interested in your work on Myocardium Motion. I have a question that the Myocardium Motion and Myocardium Strain may based on short aixs of CTA? I see your work based on normal scan. By the way, based on the Myocardium Motion, I wonder if you will release the way to compute Myocardium Strain later.

Best wishes

**Comment by shaharzuler** [comment]
Hello @mayayayahong ,

Thank you for your interest in the project!

The scan orientation does not affect the functionality of the code. You can use any orientation you prefer. The movement is computed globally in 3D, not per individual slice, so both short-axis and long-axis views are compatible. I hope this answers your question.

Regarding myocardium strain calculation: I do not currently plan to release an official strain computation method. However, I found a function from an earlier version of my code that might help you get started. Please note that I have not recently tested it, so please use it with caution.

This function estimates strain on a **point cloud** representing a **surface**. For each point, it finds the k-nearest neighbors and measures the change in variance of the first two principal components after warping. The two components are used because they capture surface motion. For voxel-based data or a point cloud that is not surface-based, further modifications will be necessary. I hope this helps!

```python

from sklearn.decomposition import PCA
from torch_cluster import knn
import numpy as np
import torch

def calc_stretch_ratio_from_flow(k_nn, flow, source_pcd) -> np.array:
    source_neigh_idxs = knn(torch.tensor(source_pcd), torch.tensor(source_pcd), k_nn)[1, :].reshape(source_pcd.shape[0], k_nn).numpy()
    strains = np.zeros(source_pcd.shape[0])

    for point_idx in range(source_pcd.shape[0]):
        nn_source_k_indices = source_neigh_idxs[point_idx, :k_nn]
        source_knn_coords = source_pcd[nn_source_k_indices]

        flow_knn_coords = flow[nn_source_k_indices]
        warped_knn_coords = source_knn_coords + flow_knn_coords

        pca_source = PCA(n_components=3).fit(source_knn_coords)
        var_2d_source = pca_source.explained_variance_[:-1]
        var_2d_source_magnitude = np.linalg.norm(var_2d_source)

        pca_target = PCA(n_components=3).fit(warped_knn_coords)
        var_2d_target = pca_target.explained_variance_[:-1]
        var_2d_target_magnitude = np.linalg.norm(var_2d_target)

        strains[point_idx] = var_2d_target_magnitude / var_2d_source_magnitude  # estimate strain as the ratio of target to source variance magnitudes

    return strains


```


Feel free to reach out if you have further questions!

Best wishes,
Shahar
